### PR TITLE
multi: support multiple RPC files

### PR DESCRIPTION
**Pull request by guggero**
Updates the API docs with the following features:
* Make compatible with `lnd` and `loop`, generic enough to also easily include other projects in the future
* Include all `*.proto` files for gRPC
* Add service name in front of RPC, sort alphabetically
* Show both gRPC and REST on the same page
* Mention which protos are considered experimental and what that means
* Link to the proto files for easy access
* Mention commit the documentation was generated from
* Link to message type definition in protos
* List Enumerations (was a contributor PR that I included)
* Dockerize everything, removing the need to install a big list of tools/libraries just to generate the documentation

Also includes the commits of #8 and #10 because they seem to never have been looked at before.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,20 +47,11 @@ You can now see the docs at `http://localhost:4567`.
 # Install Jinja for python templating.
 pip3 install Jinja2
 
-# Set your $GOPATH to a directory that has access to lncli.
-export GOPATH="/path/to/lnd"
-export PATH=$PATH:$GOPATH/bin
-
-# Install annotations and protoc-gen-doc.
-go get -u github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis
```
Comment (wpaulino) [inline README.md:55]: This is still needed, but the go command won't work. We'll want to clone the repo instead to `$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway`. We should make sure not to clone it if it already exists too.

File: update_and_render.sh [hunk 2]
```diff
@@ -2,26 +2,42 @@
...
+rsync -a --prune-empty-dirs --include '*/' --include '*.proto' --include '*.json' --exclude '*' $LND_DIR/lnrpc/ $PROTO_DIR/
 
-# Update lncli to the respective commit.
-pushd $GOPATH/src/github.com/lightningnetwork/lnd
-git reset --hard HEAD
-git pull
-git checkout $commit
-make clean && make && make install
+pushd $PROTO_DIR
+protoc -I. -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
```
Comment (wpaulino) [inline update_and_render.sh]: Needed to add an additional include here: `-I/usr/local/include` when trying this out on Linux.

File: update_and_render.sh [hunk 3]
```diff
@@ -2,26 +2,42 @@
...
-curl -o rpc.swagger.json -s https://raw.githubusercontent.com/lightningnetwork/lnd/$commit/lnrpc/rpc.swagger.json
+# Update lncli to the respective commit.
+pushd $LND_DIR
+COMMIT=$(git rev-parse HEAD)
+if [ $commit != "nocompile" ]; then
+  git reset --hard HEAD
+  git pull
+  git checkout $commit
+  COMMIT=$(git rev-parse HEAD)
+  make clean && make && make install tags="$EXPERIMENTAL_PACKAGES"
```
Comment (wpaulino) [inline update_and_render.sh]: Nit: can just do `make clean && make install`.

File: README.md (hunk 1 above)
Comment (guggero) [reply README.md:55]: I added the line back, but I think it's probably easier to just use the dockerized way in the future.

File: update_and_render.sh (hunk 2 above)
Comment (guggero) [reply update_and_render.sh]: Yes, thanks, added that.

**Comment by guggero** [comment]
I think this can be merged now? The official documentation is now generated and uploaded from this script.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,39 @@
+FROM golang:1.13-buster
```
Comment (Roasbeef) [inline Dockerfile]: 🤩

**Pull request merged**